clusterous/cluster.py

```diff
@@ -643,6 +640,161 @@ def get_cluster_spec(self):
 
         return spec
 
+    def _nat_ssh_port_forwarding(self, nat_public_ip, controller_private_ip):
+        time.sleep(45)
+        self._logger.debug('Forwarding port 22000 on NAT ({0}) to Controller ({1}) port 22'.format(nat_public_ip, controller_private_ip))
+
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(hostname = nat_public_ip,
+                    username = 'ec2-user',
+                    key_filename = os.path.expanduser(self._config['key_file']))
+        cmd = 'sudo iptables -t nat -A PREROUTING -p tcp --dport 22000 -j DNAT --to-destination {0}:22'.format(controller_private_ip)
+        # Using 'invoke_shell' because it is 'sudo' cmd
+        channel = ssh.invoke_shell()
+        channel.send('{0}\n'.format(cmd))
+        time.sleep(3)
+        output = channel.recv(99999)
+        ssh.close()
+        
+        if '000 -j DNAT' not in output:
+            message = 'Error: Setting SSH port forwarding on NAT instance.'
+            return (False, message)
+
+        return (True, '')
+
+    def _tag_resource(self, resource_name):
```

Comment:
This is a useful method. Perhaps change all the other tagging code in this class to use this method for the sake of consistency